react/DiscountBadge.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react'
+import { FormattedNumber } from 'react-intl'
+import PropTypes from 'prop-types'
+
+/**
+* The discount badge component. It receives the product's list and selling prices
+* and calculates the discount percent to show it in the product's sumary.
+*/
+class DiscountBadge extends Component {
+  calculateDiscountTax(listPrice, sellingPrice) {
+    return (listPrice - sellingPrice) / listPrice
```

Comment:
Tem que ser feito um tratamento aqui, para quando o listPrice for 0. Divisão por zero dá infinity em javascript.